src/utilities/request.jl

```
@@ -13,6 +13,31 @@ Base.@kwdef mutable struct Request
     http_options::AbstractDict{Symbol,<:Any}=LittleDict{Symbol,String}()
     return_raw::Bool=false
     response_dict_type::Type{<:AbstractDict}=LittleDict
+    downloader::Union{Nothing, Downloads.Downloader}=nothing
+end
+
+abstract type AbstractBackend end
+struct HTTPBackend <: AbstractBackend end
+struct DownloadsBackend <: AbstractBackend end
+
+default_backend() = DownloadsBackend()
+
+submit_request(aws::AbstractAWSConfig, request::Request; return_headers::Bool=false) = submit_request(default_backend(), aws, request; return_headers)
```

right now, the user changes the backend by defining
```julia
AWS.default_backend() = AWS.HTTPBackend()
```
for example.

Another option would be to use [Preferences.jl](https://github.com/JuliaPackaging/Preferences.jl) to set a preference, but that's 1.6+ only.

The word "default" here is kind of misleading because there's no other way to change it (e.g. on a per-request basis). I'm not sure if anyone would want to mix backends between different requests or not.

I think this maybe should start as an experimental feature that the user opts into, and if it works well then Downloads.jl could be promoted to the default.